server/lib/errors/NoPermission.js

```diff
@@ -0,0 +1,52 @@
+const util = require('util');
+
+/**
+ * @class NoPermission
```

Comment:
The `lib/BadRequest` class allows you to specify a key that will be sent to the user. In order to have less server code to maintain I suggest that these cases are handled as `BadRequest`s with unique error codes passed in the key field.